packages/vscode-container-client/src/clients/DockerClientBase/parsePruneLikeOutput.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { asIds } from "../../utils/asIds";
+import { tryParseSize } from "./tryParseSize";
+
+export type PruneParseOptions = {
+    spaceReclaimedRegex?: RegExp;
+    resourceRegex?: RegExp;
+};
+
+export type PruneResult = {
+    spaceReclaimed: number;
+    resources: string[];
+};
+
+export function parsePruneLikeOutput(output: string, options: PruneParseOptions): PruneResult {
+    const deletedResources: string[] = [];
+    let spaceReclaimed: number = 0;
+
+    const lines = asIds(output);
```

Comment:
`asIds` isn't meant to be used as a line-breaking feature. In fact, if you use `'foo'.match(/regex/g)` you might not need to break things up by line at all, just run it through the two regex (one for matching ID's/names of resources, one for matching the reclaimed space.

Something like this:

```typescript
const resourceMatches = output.match(/^\w+$/igm); // the G flag means global, gimme all matches. The M flag means multiline, which says to treat ^ and $ as the beginning and end of lines, not the whole string

const spaceReclaimedMatches = /Total space reclaimed: ([\w\s]+)/i.exec(output);

// Continue with getting the group out, tryParseSize, etc.
```

Reply:
yeah that makes sense